CHAID/stats.py

```diff
@@ -50,11 +50,15 @@ def best_split(self, ind, dep):
         if isinstance(dep, ContinuousColumn):
             return self.best_con_split(ind, dep)
         else:
-            return self.best_cat_split(ind, dep)
+            return self.best_cat_heuristic_split(ind, dep)
 
-    def best_cat_split(self, ind, dep):
-        """ detrmine best categorical variable split """
+    def best_cat_heuristic_split(self, ind, dep):
+        """ determine best categorical variable split using heuristic methods """
         split = Split(None, None, None, None, 0)
+        if len(np.unique(dep.arr)) == 1:
+            split.invalid_reason = InvalidSplitReason.PURE_NODE
```

Comment:
`PURE_NODE` is a bit of a bad name. 2 hard problems I guess and it is a pre existing thing though so ... 🤷‍♂️ 

Does this conditional mean we can get rid of the previous use below?

Reply:
yes, will sort